Bloom.md

```diff
@@ -0,0 +1,341 @@
+---
+RFC: 4
+Status: Proposed
+---
+
+# ValkeyBloom Module RFC
+
+## Abstract
+
+The proposed feature is ValkeyBloom which is a Rust based Module that brings a native bloom filter data type into Valkey.
+The module should be API-compatible with Redis Ltd.’s ReBloom Module.
+
+## Motivation
+
+Bloom filters are a space efficient probabilistic data structure that can be used to “check” whether an element exists in
+a set (with a defined false positive), and to “add” elements to a set. While checking whether an item exists, false positives
+are possible, but false negatives are not possible. https://en.wikipedia.org/wiki/Bloom_filter
+
+To utilize Bloom filters in their client applications, users today use client libraries that are compatible with the ReBloom
+including jedis, redis-py, node-redis, nredisstack, rueidis, rustis, etc. This allows customers to perform bloom filter
+based operations, e.g. add and set.
+
+Redis Ltd.‘s ReBloom is published under a proprietery license and hence cannot be distributed freely with ValKey.
+
+There is growing [demand](https://github.com/orgs/valkey-io/discussions?discussions_q=+bloom+) for an
+(1) Open Source bloom filter feature in Valkey which is (2) compatible with the ReBloom API syntax and with
+existing ReBloom based client libraries. ValkeyBloom will help address both these requirements.
+
+
+## Design Considerations
+
+The ValkeyBloom module brings in a bloom module data type into Valkey and provides commands to create / reserve
+bloom filters, operate on them (add items, check if items exist), inspect bloom filters, etc.
+It allows customization of properties of bloom filter (capacity, false positive rate, expansion rate, specification of
+scaling vs non-scaling filters etc) through commands and configurations. It also allows users to create scalable bloom
+filters and back up & restore bloom filters (through RDB load and save).
+
+ValkeyBloom provides commands (BF.*), configs, etc to operate on Bloom objects which are top-level structures containing
+lower-level BloomFilter provided by an external crate. ValkeyBloom utilizes an open source (BSD-2) Bloom Filter Rust library
+around which it implements a scalable bloom filter.
+
+When a bloom filter is created, a bit array is created with a length proportional to the capacity (number of items the user
+wants to add to the filter) and hash functions are also created. The number of hash functions are controlled by the
+false positive rate that the user configures.
+
+When a user adds an item (e.g. BF.ADD) to the filter, the item is passed through the filter and corresponding bit are set
+to 1. When a user checks whether an item exists on a filter (e.g. BF.EXISTS), the item is passed through the filters
+and if all the resolved bits have values as 1, we can say that the item exists with a false positive rate of
+X (specified by the user when creating the filter). If any of the bits are 0, the item does not exist and the BF.EXISTS
```

Comment:
Any external link explaining the calculation of this probability?

Reply:
This is one source: https://en.wikipedia.org/wiki/Bloom_filter#Probability_of_false_positives